test/dart2_suggestors/pubspec_over_react_upgrader_test.dart

```diff
@@ -15,10 +15,35 @@
 import 'package:over_react_codemod/src/dart2_suggestors/pubspec_over_react_upgrader.dart';
 import 'package:pub_semver/pub_semver.dart';
 import 'package:test/test.dart';
+import 'package:yaml/yaml.dart';
 
 import '../shared_pubspec_tests.dart';
 import '../util.dart';
 
+/// Throws if [yaml] is an invalid pubspec, either due to:
+///
+/// - being unparseable
+/// - having incorrect structure (this check is not comprehensive)
+void validatePubspecYaml(String yaml) {
```

Comment:
There are three copies of this now. Can we combine them into a common implementation so they stay in sync?